### DISCUSSION TITLE
Create a model plugin to calculate the energy and carbon consumed during AI model training

### DISCUSSION THREAD

**Discussion by navveenb** [Green-Software-Foundation/hack#33] [Develop an Idea]
## Story
As a user, I want to calculate the environmental impact of the AI model

## Rationale
This idea is to generate a model that could be used to calculate the energy and carbon consumed during AI model training. 

## Scope
This is a Model implementation.

## Implementation guidelines
This model MUST conform to the `ModelPlugin` interface*

## Difficulty
4/5

## Impact
4/5

## Examples and resources
[IMPL writing guide](https://if.greensoftware.foundation/tutorials/how-to-write-impls)

**Comment by sann3** [comment]
I am thinking to implement this using https://mlco2.github.io/impact/. We can get the inputs  Hardware type, Hours Used, Provider, Region of Compute and return the raw carbon emissions produced and the approximate offset carbon emissions. I know the data is outdated, but in future if we get a updated source we can use it. 

If we need more accurate results then we can also integrate with paid services like  https://www.climatiq.io
